### PR TITLE
feat: handling of subscription items and features during adjustments

### PR DESCRIPTION
**Pull request by joeysabs**
## What Does this PR Do?

---
## Summary by cubic
Added helpers to adjust subscription items: preserve manual items, expire only non-matching non-manual items, upsert items and features, and grant mid-period credits without duplication. Keeps/updates items by id, grants full credits for Once features, and posts ledger entries for grants.

- **New Features**
  - Added handleSubscriptionItemAdjustment to coordinate item expiry, upserts, and feature creation.
  - Items with id are kept/updated; items without id are created; non-matching non-manual items are expired; throws if provided id does not exist.
  - Prorates EveryBillingPeriod grants mid-period, grants full for Once, skips grants at period start, and creates ledger entries.
  - Expires manual features that overlap with plan features (plan wins).

<sup>Written for commit 8543f51c3f295be6ad0be8798be3d1adcf8ddf71. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Public subscription-adjustment API and utilities for handling item activity and manual vs plan items; mid-period proration and full-credit handling with automatic credit and ledger entries, and rule-driven feature creation/expiration.

* **Bug Fixes**
  * Stronger validation on item updates to prevent updating non-existent items (now surfaces an error).

* **Tests**
  * Large end-to-end test suite covering adjustments, credit calculations, feature creation/expiration, ledger entries, and many edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.ts [hunk 1]
```diff
@@ -41,3 +82,402 @@ export const isSubscriptionItemActiveAndNonManual = (
...
+        overlappingManualFeatures.map((feature) =>
+          expireSubscriptionItemFeature(
+            feature,
+            adjustmentDate,
+            transaction
+          )
+        )
+      )
+    }
+  }
```
Comment (agreea) [inline platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.ts]: This is so subtle and specific in its intent that we should document exactly why we do it this way, and why we think it makes sense for all of:
- overlapping toggles
- overlapping usage credit grants (of exactly the same parent feature)

we may want to leave a note that it's possible some may find this confusing and that we should be prepared to adjust whether we expire overlapping manually created credit grants.
